
<launch>
    <arg name="camera_info_topic"               default="camera/depth/camera_info"/>
    <arg name="depth_rect_raw_topic"            default="camera/depth/image_rect_raw"/>

    <arg name="calibrated_depth_topic"          value="sensor_3d/depth/calibration/depth_calibrated"/>
    <arg name="calibrated_cloud_topic"          value="sensor_3d/depth/calibration/points"/>

    <arg name="camera_info_relay_topic"         value="sensor_3d/depth/calibration/camera_info"/>
    <arg name="depth_relay_topic"               value="sensor_3d/depth/calibration/depth_relay"/>
    <arg name="save_calibration_trigger_topic"  value="sensor_3d/depth/calibration/save_calibration"/>

    <node pkg="nodelet" type="nodelet" name="cloudify_manager" args="manager"/>	
    <node pkg="nodelet" type="nodelet" name="cloudify_calibration" args="load depth_image_proc/point_cloud_xyz cloudify_manager">
        <remap from="image_rect"    to="$(arg calibrated_depth_topic)"/>
        <remap from="points"        to="$(arg calibrated_cloud_topic)"/>
    </node>


    <node pkg="depth_calibration" type="depth_calibrator" name="depth_calibrator" output="screen">
        <remap from="camera_info_relay"         to="$(arg camera_info_relay_topic)"/>
        <remap from="depth_relay"               to="$(arg depth_relay_topic)"/>
        
        <remap from="calibrated_depth"          to="$(arg calibrated_depth_topic)"/>
        <remap from="calibrated_cloud"          to="$(arg calibrated_cloud_topic)"/>
        <remap from="save_calibration_trigger"  to="$(arg save_calibration_trigger_topic)"/>
    </node>

    <node pkg="depth_calibration" type="depth_calibration.py" name="depth_calibration" output="screen">	  
        <remap from="camera_info_relay"         to="$(arg camera_info_relay_topic)"/>
        <remap from="depth_relay"               to="$(arg depth_relay_topic)"/>
        <remap from="save_calibration_trigger"  to="$(arg save_calibration_trigger_topic)"/>
        
        <remap from="camera_info"               to="$(arg camera_info_topic)"/>
        <remap from="depth_rect_raw"            to="$(arg depth_rect_raw_topic)"/>
        <remap from="calibrated_cloud"          to="$(arg calibrated_cloud_topic)"/>
    </node>
</launch>
